corehq/apps/userreports/expressions/specs.py

```diff
@@ -263,3 +266,31 @@ def get_variables(self, item, context):
             for slug, variable_expression in self._context_variables.items()
         }
         return var_dict
+
+
+class FormsExpressionSpec(JsonObject):
+    type = TypeProperty('get_case_forms')
+    case_id_expression = DefaultProperty(required=True)
+
+    def configure(self, case_id_expression):
+        self._case_id_expression = case_id_expression
+
+    def __call__(self, item, context=None):
+        case_id = self._case_id_expression(item, context)
+
+        if not case_id:
+            return []
+
+        assert context.root_doc['domain']
+        return self._get_forms(case_id, context.root_doc['domain'])
+
+    def _get_forms(self, case_id, domain):
+        # todo, cahcing
+        try:
+            case = CaseAccessors(domain).get_case(case_id)
+            assert case.domain == domain
+        except (CaseNotFound, AssertionError):
+            return []
+
+        xforms = FormProcessorInterface(domain).get_case_forms(case_id)
```

Comment:
If any one form is not from the same domain, should we exclude it and return the rest of forms or return empty list altogether? (Realistically. will forms for one case be ever from two different domains?